saleor/graphql/order/bulk_mutations/order_bulk_create.py

```diff
@@ -518,6 +543,132 @@ class Meta:
         doc_category = DOC_CATEGORY_ORDERS
         error_type_class = OrderBulkCreateError
 
+    @classmethod
+    def get_all_instances(cls, orders_input) -> Dict[str, Any]:
+        """Retrieve all required instances to process orders.
+
+        Return:
+            Dictionary with keys "{model_name}.{key_name}.{key_value}" and model
+            instances as values.
+
+        """
+        # Collect all model keys from input
+        keys = defaultdict(list)
+        for order in orders_input:
+            keys["User.id"].append(order["user"].get("id"))
+            keys["User.email"].append(order["user"].get("email"))
+            keys["User.external_reference"].append(
+                order["user"].get("external_reference")
+            )
+            keys["Channel.slug"].append(order.get("channel"))
+            keys["Voucher.code"].append(order.get("voucher"))
+            keys["Warehouse.id"].append(order["delivery_method"].get("warehouse_id"))
+            keys["ShippingMethod.id"].append(
+                order["delivery_method"].get("shipping_method_id")
+            )
+            keys["TaxClass.id"].append(
+                order["delivery_method"].get("shipping_tax_class_id")
+            )
+            keys["Order.number"].append(order.get("number"))
+            for note in order["notes"]:
+                keys["User.id"].append(note.get("user_id"))
+                keys["User.email"].append(note.get("user_email"))
+                keys["User.external_reference"].append(
+                    note.get("user_external_reference")
+                )
+                keys["App.id"].append(note.get("app_id"))
+            for order_line in order["lines"]:
+                keys["ProductVariant.id"].append(order_line.get("variant_id"))
+                keys["ProductVariant.sku"].append(order_line.get("variant_sku"))
+                keys["ProductVariant.external_reference"].append(
+                    order_line.get("variant_external_reference")
+                )
+                keys["Warehouse.id"].append(order_line.get("warehouse"))
+                keys["TaxClass.id"].append(order_line.get("tax_class_id"))
+            for fulfillment in order["fulfillments"]:
+                for line in fulfillment["lines"]:
+                    keys["ProductVariant.id"].append(line.get("variant_id"))
+                    keys["ProductVariant.sku"].append(line.get("variant_sku"))
+                    keys["ProductVariant.external_reference"].append(
+                        line.get("variant_external_reference")
+                    )
+                    keys["Warehouse.id"].append(line.get("warehouse"))
+            for gift_card_code in order["gift_cards"]:
+                keys["GiftCard.code"].append(gift_card_code)
+
+        # Convert global ids to model ids and get rid of Nones
+        for key, values in keys.items():
+            keys[key] = [value for value in values if value is not None]
+            if key.split(".")[1] == "id":
```

Comment:
If you always have a structure of `key` like `<type>.<identifier>`, You could do something like this 👇  Then you don't need to call `key.split(".")[0]` for each global_id in values.
```
object_type, key_field = key.split(".")
if key_field == "id":
    model_ids = []
    for global_id in values:
        try:
            _, id = from_global_id_or_error(
                str(global_id), object_type, raise_error=True
            )
            model_ids.append(id)
        except GraphQLError:
            pass
    keys[key] = model_ids
```